### DISCUSSION TITLE
Iterative DPO

### DISCUSSION THREAD

**Discussion by Sanqiang** [lucidrains/self-rewarding-lm-pytorch#6] [Q&A]
I think authors of self rewarding llm didn't use standard DPO but Interactive DPO, which is from their another paper: https://arxiv.org/pdf/2312.16682.pdf.

**Comment by lucidrains** [comment]
<img width="592" alt="Screen Shot 2024-01-25 at 3 24 48 PM" src="https://github.com/lucidrains/self-rewarding-lm-pytorch/assets/108653/92806775-ca9b-4a55-8310-e11b1bc5483a">

is this the section you are looking at? they say it is similar to the cringe paper, but then goes on to outline what they actually did

**Reply by lucidrains** [reply]
does the [following code](https://github.com/lucidrains/self-rewarding-lm-pytorch/blob/main/self_rewarding_lm_pytorch/self_rewarding_lm_pytorch.py#L643-L657) look in line with what is described in the screenshot above?

**Reply by lucidrains** [reply]
@Sanqiang you mean "Iterative", not "Interactive"?

**Reply by Minami-su** [reply]
Your code is correct.

In fact, the author is suggesting that this self-reward method is similar to PCO, as described in this paper: https://arxiv.org/pdf/2312.16682.pdf.
